packages/browser-integration-tests/src/custom-global-key.test.ts

```diff
@@ -0,0 +1,53 @@
+import { test, expect } from '@playwright/test'
+import { standaloneMock } from './helpers/standalone-mock'
+import { extractWriteKeyFromUrl } from './helpers/extract-writekey'
+import { CDNSettingsBuilder } from '@internal/test-helpers'
+
+test.describe('Segment with custom global key', () => {
+  test.beforeEach(standaloneMock)
+  test.beforeEach(async ({ context }) => {
+    await context.route(
+      'https://cdn.segment.com/v1/projects/*/settings',
+      (route, request) => {
+        if (request.method().toLowerCase() !== 'get') {
+          return route.continue()
+        }
+
+        const writeKey = extractWriteKeyFromUrl(request.url()) || 'writeKey'
+        return route.fulfill({
+          status: 200,
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(new CDNSettingsBuilder({ writeKey }).build()),
+        })
+      }
+    )
+  })
+
+  test('supports using a custom global key', async ({ page }) => {
+    // Load analytics.js
+    await page.goto('/standalone-custom-key.html')
+    await page.evaluate(() =>
+      (window as any).segment_analytics.load('fake-key')
```

Comment:
Ah, so if a custom key is used, we don't actually have to set `globalAnalyticsKey`? That's nice!

Reply:
Yes, it serves no purpose for snippet users. Unless it serves a utility for NPM users, it is useless.